// angular5-time-picker/src/time-control/w-time-dialog/w-time-dialog.component.js
import { Component, Inject } from '@angular/core';
import { MatDialogRef } from '@angular/material';
import { MAT_DIALOG_DATA } from '@angular/material';
import { CLOCK_TYPE } from '../w-clock/w-clock.component';
var WTimeDialogComponent = /** @class */ (function () {
    function WTimeDialogComponent(
    //@Inject(MAT_DIALOG_DATA) private data: { time: ITime, color: string },
    data, color, dialogRef) {
        this.data = data;
        this.color = color;
        this.dialogRef = dialogRef;
        this.VIEW_HOURS = CLOCK_TYPE.HOURS;
        this.VIEW_MINUTES = CLOCK_TYPE.MINUTES;
        this.currentView = this.VIEW_HOURS;
        this.userTime = data.time;
        this.color = data.color;
        console.log('this.color', this.color);
    }
    WTimeDialogComponent.prototype.revert = function () {
        this.dialogRef.close(-1);
    };
    WTimeDialogComponent.prototype.submit = function () {
        this.dialogRef.close(this.userTime);
    };
    WTimeDialogComponent.decorators = [
        { type: Component, args: [{
                    selector: 'w-time-dialog',
                    template: "\n    <div mat-dialog-content class=\"w-timepicker-dialog\">\n      <w-time [color]=\"color\" [userTime]=\"userTime\" (onRevert)=\"revert()\" (onSubmit)=\"submit()\"></w-time>\n    </div>",
                    styles: ["\n    .w-timepicker-dialog {\n      padding: 0;\n      margin: -24px;\n      width: calc(100% + 48px);\n    }\n    "]
                },] },
    ];
    /** @nocollapse */
    WTimeDialogComponent.ctorParameters = function () { return [
        { type: undefined, decorators: [{ type: Inject, args: [MAT_DIALOG_DATA,] },] },
        { type: undefined, decorators: [{ type: Inject, args: [MAT_DIALOG_DATA,] },] },
        { type: MatDialogRef, },
    ]; };
    return WTimeDialogComponent;
}());
export { WTimeDialogComponent };
