// ng-http-loader/esm2015/lib/services/spinner-visibility.service.js
import { Injectable } from '@angular/core';
import { ReplaySubject } from 'rxjs';
import { PendingInterceptorService } from './pending-interceptor.service';
import * as i0 from "@angular/core";
import * as i1 from "./pending-interceptor.service";
export class SpinnerVisibilityService {
    constructor(pendingInterceptorService) {
        this.pendingInterceptorService = pendingInterceptorService;
        this._visibilitySubject = new ReplaySubject(1);
    }
    get visibilityObservable() {
        return this._visibilitySubject.asObservable();
    }
    show() {
        this.pendingInterceptorService.forceByPass = true;
        this._visibilitySubject.next(true);
    }
    hide() {
        this._visibilitySubject.next(false);
        this.pendingInterceptorService.forceByPass = false;
    }
}
SpinnerVisibilityService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] },
];
SpinnerVisibilityService.ctorParameters = () => [
    { type: PendingInterceptorService, },
];
/** @nocollapse */ SpinnerVisibilityService.ngInjectableDef = i0.defineInjectable({ factory: function SpinnerVisibilityService_Factory() { return new SpinnerVisibilityService(i0.inject(i1.PendingInterceptorService)); }, token: SpinnerVisibilityService, providedIn: "root" });
function SpinnerVisibilityService_tsickle_Closure_declarations() {
    SpinnerVisibilityService.decorators;
    SpinnerVisibilityService.ctorParameters;
    SpinnerVisibilityService.prototype._visibilitySubject;
    SpinnerVisibilityService.prototype.pendingInterceptorService;
}
